rosdiagnostic/src/rosdiagnostic/rosdiagnostic.py

```diff
@@ -0,0 +1,234 @@
+
+    parser.add_argument('--namespace', action='store', metavar='NAMESPACE', type=str, default='~', dest='ns',
+                        help='root namespace where to look for the "/diagnostics_agg" topic')
+
+    parser.add_argument('--filter', action='store', metavar='FILTER', type=str, default='.*', dest='filter',
+                        help='regular expression to be applied as a filter to the diagnostic name')
+
+    parser.add_argument('--nocolor', action='store_true',
+                        help='output should not make use of any color')
+
+    parser.add_argument('--detail', action='store_true',
+                        help='printing the full diagnostic details ')
+
+    args = parser.parse_args(argv)
+
+    _check_master()
+    rospy.init_node('rosdiagnostic', anonymous=True)
+    rosdiag = RosDiagnostics(args)
+    rospy.spin()
+    del rosdiag
+
+
+def _rosdiagnostic_cmd_version(prog):
+    stack_name = os.path.basename(__file__)
+    rosstack = rospkg.RosStack()
+    try:
+        version = rosstack.get_stack_version(stack_name)
+    except rospkg.ResourceNotFound:
+        try:
+            # hack to make it work with wet packages
+            mm = ManifestManager(PACKAGE_FILE)
+            path = mm.get_path(stack_name)
+            package_manifest = os.path.join(path, 'package.xml')
+            if os.path.exists(package_manifest):
+                from xml.etree.ElementTree import ElementTree
+                try:
+                    root = ElementTree(None, package_manifest)
+                    version = root.findtext('version')
+                except Exception:
+                    pass
+        except rospkg.ResourceNotFound:
+            version = '(unknown)'
+
+    print("{} {}".format(prog, version))
+    return version
+
+
+def _fullusage():
+    print("""rosdiagnostic is a command-line tool for printing information about ROS Diagnostics.
+
+Commands:
+\trosdiagnostic echo\tprint diagnostic messages to screen
+\trosdiagnostic version\tprint the version of this application
+
+Type rosdiagnostic <command> -h for more detailed usage, e.g. 'rosdiagnostic echo -h'
+""")
+    sys.exit(getattr(os, 'EX_USAGE', 1))
+
+
+def rosdiagnosticmain(argv=None):
+    if argv is None:
+        argv = sys.argv
+    # filter out remapping arguments in case we are being invoked via roslaunch
+    argv = rospy.myargv(argv)
+
+    # process argv
+    if len(argv) == 1:
+        _fullusage()
+    try:
+        command = argv[1]
+        if command == 'echo':
+            _rosdiagnostic_cmd_echo(argv[2:])
+        elif command == 'version':
+            _rosdiagnostic_cmd_version('rosdiagnostic')
+        else:
+            _fullusage()
+    except socket.error:
+        sys.stderr.write("Network communication failed. Most likely failed to communicate with master.\n")
+        sys.exit(1)
+    except rosbag.ROSBagException as e:
+        sys.stderr.write("ERROR: unable to use bag file: %s\n" % str(e))
+        sys.exit(1)
```

Comment:
This catch for rosbag exceptions seems unnecessary; I don't see `rosbag` used anywhere else in this module, or any command-line arguments for specifying a bag file.